diff --git a/build/modules/indy/services/IndyRevocationService.js b/build/modules/indy/services/IndyRevocationService.js
index 0c5346f989ae234afb93a90432f1310cb897cd25..14f7d830cdb4fc5b419e53aa67929474e65e4c4c 100644
--- a/build/modules/indy/services/IndyRevocationService.js
+++ b/build/modules/indy/services/IndyRevocationService.js
@@ -121,10 +121,10 @@ let IndyRevocationService = class IndyRevocationService {
         if (!requestRevocationInterval.to) {
             throw new AriesFrameworkError_1.AriesFrameworkError(`Presentation requests proof of non-revocation with no 'to' value specified`);
         }
-        if ((requestRevocationInterval.from || requestRevocationInterval.from === 0) &&
-            requestRevocationInterval.to !== requestRevocationInterval.from) {
-            throw new AriesFrameworkError_1.AriesFrameworkError(`Presentation requests proof of non-revocation with an interval from: '${requestRevocationInterval.from}' that does not match the interval to: '${requestRevocationInterval.to}', as specified in Aries RFC 0441`);
-        }
+        // if ((requestRevocationInterval.from || requestRevocationInterval.from === 0) &&
+        //     requestRevocationInterval.to !== requestRevocationInterval.from) {
+        //     throw new AriesFrameworkError_1.AriesFrameworkError(`Presentation requests proof of non-revocation with an interval from: '${requestRevocationInterval.from}' that does not match the interval to: '${requestRevocationInterval.to}', as specified in Aries RFC 0441`);
+        // }
     }
 };
 IndyRevocationService = __decorate([
\ No newline at end of file
