diff --git a/lib/commonjs/components/views/KeyboardView.js b/lib/commonjs/components/views/KeyboardView.js
index 70e2d62c1185bb1a8c0e0f1d56e1a69a9554f4de..159169608c56ac12dd700ab72ece4873fc293281 100644
--- a/lib/commonjs/components/views/KeyboardView.js
+++ b/lib/commonjs/components/views/KeyboardView.js
@@ -30,7 +30,8 @@ const useSafeHeaderHeight = () => {
  */
 const KeyboardView = ({
   children,
-  scrollViewProps
+  scrollViewProps,
+  scrollViewRef
 }) => {
   const safeHeaderHeight = useSafeHeaderHeight();
   return /*#__PURE__*/_react.default.createElement(_reactNative.KeyboardAvoidingView, {
@@ -44,7 +45,8 @@ const KeyboardView = ({
     contentContainerStyle: [{
       flexGrow: 1
     }, scrollViewProps === null || scrollViewProps === void 0 ? void 0 : scrollViewProps.contentContainerStyle],
-    showsVerticalScrollIndicator: false
+    showsVerticalScrollIndicator: false,
+    ref: scrollViewRef
   }, scrollViewProps), children));
 };
 var _default = exports.default = KeyboardView;
diff --git a/lib/commonjs/components/views/ScreenWrapper.js b/lib/commonjs/components/views/ScreenWrapper.js
index dd0a15008d72326c494d097b4dae1689f5a86bd7..a66d44d910417baf2e49f8ed4878f7723fac2d9e 100644
--- a/lib/commonjs/components/views/ScreenWrapper.js
+++ b/lib/commonjs/components/views/ScreenWrapper.js
@@ -22,7 +22,8 @@ const ScreenWrapper = ({
   scrollable = true,
   scrollViewContainerStyle,
   controlsContainerStyle,
-  padded = true
+  padded = true,
+  scrollViewRef
 }) => {
   const {
     Spacing,
@@ -53,7 +54,8 @@ const ScreenWrapper = ({
     }
     return /*#__PURE__*/_react.default.createElement(_reactNative.ScrollView, {
       showsVerticalScrollIndicator: false,
-      contentContainerStyle: scrollStyle
+      contentContainerStyle: scrollStyle,
+      ref: scrollViewRef
     }, children);
   };
 
@@ -63,7 +65,9 @@ const ScreenWrapper = ({
     return /*#__PURE__*/_react.default.createElement(_reactNativeSafeAreaContext.SafeAreaView, {
       style: [styles.container, style],
       edges: edges
-    }, /*#__PURE__*/_react.default.createElement(_KeyboardView.default, null, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
+    }, /*#__PURE__*/_react.default.createElement(_KeyboardView.default, {
+      scrollViewRef: scrollViewRef
+    }, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
       style: scrollStyle
     }, children), controls && /*#__PURE__*/_react.default.createElement(_reactNative.View, {
       style: [controlsStyle, {
diff --git a/lib/module/components/views/KeyboardView.js b/lib/module/components/views/KeyboardView.js
index 3fd30b5d2ec3b368741e8e22720c0f669f196b3a..407ac404dd55b1a4c87190acf83f599f19cdcbe7 100644
--- a/lib/module/components/views/KeyboardView.js
+++ b/lib/module/components/views/KeyboardView.js
@@ -23,7 +23,8 @@ const useSafeHeaderHeight = () => {
  */
 const KeyboardView = ({
   children,
-  scrollViewProps
+  scrollViewProps,
+  scrollViewRef
 }) => {
   const safeHeaderHeight = useSafeHeaderHeight();
   return /*#__PURE__*/React.createElement(KeyboardAvoidingView, {
@@ -37,7 +38,8 @@ const KeyboardView = ({
     contentContainerStyle: [{
       flexGrow: 1
     }, scrollViewProps === null || scrollViewProps === void 0 ? void 0 : scrollViewProps.contentContainerStyle],
-    showsVerticalScrollIndicator: false
+    showsVerticalScrollIndicator: false,
+    ref: scrollViewRef
   }, scrollViewProps), children));
 };
 export default KeyboardView;
diff --git a/lib/module/components/views/ScreenWrapper.js b/lib/module/components/views/ScreenWrapper.js
index 7a0e44129893f7f5e7ab18ff0a3d0e4eaaeb1124..ab9d0032c1de3d99f10373d828cb6444e25f4c1e 100644
--- a/lib/module/components/views/ScreenWrapper.js
+++ b/lib/module/components/views/ScreenWrapper.js
@@ -15,7 +15,8 @@ const ScreenWrapper = ({
   scrollable = true,
   scrollViewContainerStyle,
   controlsContainerStyle,
-  padded = true
+  padded = true,
+  scrollViewRef
 }) => {
   const {
     Spacing,
@@ -46,7 +47,8 @@ const ScreenWrapper = ({
     }
     return /*#__PURE__*/React.createElement(ScrollView, {
       showsVerticalScrollIndicator: false,
-      contentContainerStyle: scrollStyle
+      contentContainerStyle: scrollStyle,
+      ref: scrollViewRef
     }, children);
   };
 
@@ -56,7 +58,9 @@ const ScreenWrapper = ({
     return /*#__PURE__*/React.createElement(SafeAreaView, {
       style: [styles.container, style],
       edges: edges
-    }, /*#__PURE__*/React.createElement(KeyboardView, null, /*#__PURE__*/React.createElement(View, {
+    }, /*#__PURE__*/React.createElement(KeyboardView, {
+      scrollViewRef: scrollViewRef
+    }, /*#__PURE__*/React.createElement(View, {
       style: scrollStyle
     }, children), controls && /*#__PURE__*/React.createElement(View, {
       style: [controlsStyle, {
diff --git a/lib/typescript/src/components/views/KeyboardView.d.ts b/lib/typescript/src/components/views/KeyboardView.d.ts
index fd5c211b38d7053a4f00be21a490384c0abf156e..e82b58e5ecf2c2a7badec8c4fb7ed83c340e3064 100644
--- a/lib/typescript/src/components/views/KeyboardView.d.ts
+++ b/lib/typescript/src/components/views/KeyboardView.d.ts
@@ -1,5 +1,5 @@
-import React from 'react';
-import { ScrollViewProps } from 'react-native';
+import React, { RefObject } from 'react';
+import { ScrollView, ScrollViewProps } from 'react-native';
 /**
  * A wrapper component that provides keyboard-aware scrolling and safe area handling
  *
@@ -13,6 +13,7 @@ import { ScrollViewProps } from 'react-native';
 declare const KeyboardView: React.FC<{
     children: React.ReactNode;
     scrollViewProps?: ScrollViewProps;
+    scrollViewRef?: RefObject<ScrollView | null>;
 }>;
 export default KeyboardView;
 //# sourceMappingURL=KeyboardView.d.ts.map
\ No newline at end of file
diff --git a/lib/typescript/src/components/views/ScreenWrapper.d.ts b/lib/typescript/src/components/views/ScreenWrapper.d.ts
index 5334242714b85289470937dee02060ce6a80e47e..52241dc086b8c119bff9673ab37c9a19dc04500f 100644
--- a/lib/typescript/src/components/views/ScreenWrapper.d.ts
+++ b/lib/typescript/src/components/views/ScreenWrapper.d.ts
@@ -1,4 +1,4 @@
-import React, { ComponentProps } from 'react';
+import React, { ComponentProps, RefObject } from 'react';
 import { ScrollView, StyleProp, ViewStyle } from 'react-native';
 import { Edges } from 'react-native-safe-area-context';
 interface ScreenWrapperProps {
@@ -36,6 +36,10 @@ interface ScreenWrapperProps {
      * @default true
      */
     padded?: boolean;
+    /**
+     * Ref for the ScrollView. This allows parent components to control scrolling behaviour
+     */
+    scrollViewRef?: RefObject<ScrollView | null>;
 }
 /**
  * Wraps content in a SafeAreaView and optionally a KeyboardView, and provides a container for controls.
