diff --git a/App/screens/ProofRequest.tsx b/App/screens/ProofRequest.tsx
index 71e1e4247696b21925dd9757a59325cd622fcad8..e5d53781aa6c66925bfbdd2db911ae937f0adfff 100644
--- a/App/screens/ProofRequest.tsx
+++ b/App/screens/ProofRequest.tsx
@@ -68,6 +68,7 @@ const ProofRequest: React.FC<ProofRequestProps> = ({ navigation, route }) => {
   const [containsPI, setContainsPI] = useState(false)
   const [activeCreds, setActiveCreds] = useState<ProofCredentialItems[]>([])
   const [selectedCredentials, setSelectedCredentials] = useState<string[]>([])
+  const [isCredentialDisabled, setIsCredentialDisabled] = useState(true);
   const [store, dispatch] = useStore()
   const credProofPromise = useAllCredentialsForProof(proofId)
   const proofConnectionLabel = useMemo(
@@ -128,6 +129,15 @@ const ProofRequest: React.FC<ProofRequestProps> = ({ navigation, route }) => {
     },
   })
 
+  useEffect(() => {
+    setIsCredentialDisabled(
+      !hasAvailableCredentials ||
+      !hasSatisfiedPredicates(getCredentialsFields()) ||
+      revocationOffense ||
+      proof?.state !== ProofState.RequestReceived
+    );
+  }, [retrievedCredentials, proof?.state])
+
   useEffect(() => {
     const shouldShowTour = enableToursConfig && store.tours.enableTours && !store.tours.seenProofRequestTour
 
@@ -507,12 +517,7 @@ const ProofRequest: React.FC<ProofRequestProps> = ({ navigation, route }) => {
             testID={testIdWithKey('Share')}
             buttonType={ButtonType.Primary}
             onPress={handleAcceptPress}
-            disabled={
-              !hasAvailableCredentials ||
-              !hasSatisfiedPredicates(getCredentialsFields()) ||
-              revocationOffense ||
-              proof?.state !== ProofState.RequestReceived
-            }
+            disabled={isCredentialDisabled}
           />
         </View>
         <View style={styles.footerButton}>
