diff --git a/App/Theme.ts b/App/Theme.ts
index af14051..41d62f9 100644
--- a/App/Theme.ts
+++ b/App/Theme.ts
@@ -13,10 +13,10 @@ interface ColorTheme {
 }
 
 export const Colors: ColorTheme = {
-  mainColor: '#35823f',
+  mainColor: '#003366',
   activeMain: '#003366B3',
-  textColor: '#fff',
-  backgroundColor: '#000',
+  textColor: '#313132',
+  backgroundColor: '#F2F2F2',
   shadow: '#1c1c1e',
   white: '#ffffff',
   green: '#2d6e35',
diff --git a/App/globalStyles.ts b/App/globalStyles.ts
index 5f2896d..8ea0c27 100644
--- a/App/globalStyles.ts
+++ b/App/globalStyles.ts
@@ -1,6 +1,6 @@
-export const mainColor = '#35823f'
-export const textColor = '#fff'
-export const backgroundColor = '#000'
+export const mainColor = '#003366'
+export const textColor = '#313132'
+export const backgroundColor = '#F2F2F2'
 export const shadow = '#1c1c1e'
 export const borderRadius = 5
 export const green = '#2d6e35'
