diff --git a/App/theme.ts b/App/theme.ts
index 49677b4..eaad787 100644
--- a/App/theme.ts
+++ b/App/theme.ts
@@ -76,14 +76,14 @@ export const zeroOpacity = 0.0
 export const borderWidth = 2
 
 const BrandColors: BrandColors = {
-  primary: '#42803E',
-  primaryDisabled: `rgba(53, 130, 63, ${lightOpacity})`,
+  primary: '#003366',
+  primaryDisabled: `rgba(0, 51, 102, ${lightOpacity})`,
   secondary: '#FFFFFFFF',
-  secondaryDisabled: `rgba(53, 130, 63, ${heavyOpacity})`,
+  secondaryDisabled: `rgba(0, 51, 102, ${heavyOpacity})`,
   highlight: '#FCBA19',
-  primaryBackground: '#000000',
-  secondaryBackground: '#313132',
-  link: '#FFFFFF',
+  primaryBackground: '#F2F2F2',
+  secondaryBackground: '#FFFFFF',
+  link: '#1A5A96',
 }
 
 const SemanticColors: SemanticColors = {
@@ -93,22 +93,22 @@ const SemanticColors: SemanticColors = {
 }
 
 const NotificationColors: NotificationColors = {
-  success: '#313132',
-  successBorder: '#2E8540',
-  successIcon: '#2E8540',
-  successText: '#FFFFFF',
-  info: '#313132',
-  infoBorder: '#0099FF',
-  infoIcon: '#0099FF',
-  infoText: '#FFFFFF',
-  warn: '#313132',
-  warnBorder: '#FCBA19',
-  warnIcon: '#FCBA19',
-  warnText: '#FFFFFF',
-  error: '#313132',
-  errorBorder: '#D8292F',
-  errorIcon: '#D8292F',
-  errorText: '#FFFFFF',
+  success: '#DFF0D8',
+  successBorder: '#D6E9C6',
+  successIcon: '#2D4821',
+  successText: '#2D4821',
+  info: '#D9EAF7',
+  infoBorder: '#B9CEDE',
+  infoIcon: '#313132',
+  infoText: '#313132',
+  warn: '#F9F1C6',
+  warnBorder: '#FAEBCC',
+  warnIcon: '#6C4A00',
+  warnText: '#6C4A00',
+  error: '#F2DEDE',
+  errorBorder: '#EBCCD1',
+  errorIcon: '#A12622',
+  errorText: '#A12622',
 }
 
 const GrayscaleColors: GrayscaleColors = {
@@ -129,39 +129,46 @@ export const ColorPallet: ColorPallet = {
 
 export const TextTheme: TextTheme = {
   headingOne: {
+    fontFamily: 'BCSans-Regular',
     fontSize: 38,
     fontWeight: 'bold',
-    color: ColorPallet.grayscale.white,
+    color: ColorPallet.grayscale.darkGrey,
   },
   headingTwo: {
+    fontFamily: 'BCSans-Regular',
     fontSize: 32,
     fontWeight: 'bold',
-    color: ColorPallet.grayscale.white,
+    color: ColorPallet.grayscale.darkGrey,
   },
   headingThree: {
+    fontFamily: 'BCSans-Regular',
     fontSize: 26,
     fontWeight: 'bold',
-    color: ColorPallet.grayscale.white,
+    color: ColorPallet.grayscale.darkGrey,
   },
   headingFour: {
+    fontFamily: 'BCSans-Regular',
     fontSize: 21,
     fontWeight: 'bold',
-    color: ColorPallet.grayscale.white,
+    color: ColorPallet.grayscale.darkGrey,
   },
   normal: {
+    fontFamily: 'BCSans-Regular',
     fontSize: 18,
     fontWeight: 'normal',
-    color: ColorPallet.grayscale.white,
+    color: ColorPallet.grayscale.darkGrey,
   },
   label: {
+    fontFamily: 'BCSans-Regular',
     fontSize: 14,
     fontWeight: 'bold',
-    color: ColorPallet.grayscale.white,
+    color: ColorPallet.grayscale.darkGrey,
   },
   caption: {
+    fontFamily: 'BCSans-Regular',
     fontSize: 14,
     fontWeight: 'normal',
-    color: ColorPallet.grayscale.white,
+    color: ColorPallet.grayscale.darkGrey,
   },
 }
 
