diff --git a/App/theme.ts b/App/theme.ts
index 9eb74c1..66bea94 100644
--- a/App/theme.ts
+++ b/App/theme.ts
@@ -81,31 +81,31 @@ interface SingleSelectBlockTheme {
 
 export const Colors: ColorTheme = {
   accent: BaseColors.yellow,
-  background: BaseColors.black,
-  backgroundLight: BaseColors.lightGreen,
-  borderLight: BaseColors.mediumGreen,
-  primary: BaseColors.darkGreen,
-  primaryActive: BaseColors.darkGreenHeavyTransparent,
+  background: BaseColors.offWhite,
+  backgroundLight: BaseColors.lightBlue,
+  borderLight: BaseColors.mediumBlue,
+  primary: BaseColors.darkBlue,
+  primaryActive: BaseColors.darkBlueHeavyTransparent,
   shadow: BaseColors.darkGrey,
-  text: BaseColors.white,
+  text: BaseColors.darkGreyAlt,
   ...BaseColors,
   ...StatusColors,
 }
 
 export const CredentialTheme: CredentialTheme = {
-  background: Colors.shadow,
+  background: Colors.lightBlue,
 }
 
 export const ContactTheme: ContactTheme = {
-  background: Colors.shadow,
+  background: Colors.lightBlue,
 }
 
 export const SettingsTheme: SettingsTheme = {
-  background: Colors.shadow,
+  background: Colors.lightBlue,
 }
 
 export const SingleSelectBlockTheme: SingleSelectBlockTheme = {
-  background: Colors.shadow,
+  background: Colors.lightBlue,
 }
 
 export const TextTheme: TextTheme = {
@@ -155,7 +155,7 @@ export const Buttons = StyleSheet.create({
   primaryDisabled: {
     padding: 16,
     borderRadius: 4,
-    backgroundColor: Colors.darkGreenHeavyTransparent,
+    backgroundColor: Colors.darkBlueHeavyTransparent,
   },
   primaryText: {
     ...TextTheme.normal,
@@ -180,7 +180,7 @@ export const Buttons = StyleSheet.create({
     padding: 16,
     borderRadius: 4,
     borderWidth: 2,
-    borderColor: Colors.darkGreenLightTransparent,
+    borderColor: Colors.darkBlueLightTransparent,
     backgroundColor: Colors.white,
   },
   secondaryText: {
@@ -192,7 +192,7 @@ export const Buttons = StyleSheet.create({
   secondaryTextDisabled: {
     ...TextTheme.normal,
     fontWeight: 'bold',
-    color: Colors.darkGreenLightTransparent,
+    color: Colors.darkBlueLightTransparent,
     textAlign: 'center',
   },
 })
