diff --git a/App/theme.ts b/App/theme.ts
index f5b66aa..dbf0574 100644
--- a/App/theme.ts
+++ b/App/theme.ts
@@ -74,11 +74,11 @@ export const zeroOpacity = 0.0
 export const borderWidth = 2
 
 const BrandColors: BrandColors = {
-  primary: '#42803E',
+  primary: '#003366',
   highlight: '#FCBA19',
-  primaryBackground: '#000000',
-  secondaryBackground: '#313132',
-  link: '#FFFFFF',
+  primaryBackground: '#F2F2F2',
+  secondaryBackground: '#FFFFFF',
+  link: '#1A5A96',
 }
 
 const SemanticColors: SemanticColors = {
@@ -88,22 +88,22 @@ const SemanticColors: SemanticColors = {
 }
 
 const NotificationColors: NotificationColors = {
-  success: '#313132',
-  successBorder: '#2E8540',
-  successIcon: '#2E8540',
-  successText: '#FFFFFF',
-  info: '#313132',
-  infoBorder: '#0099FF',
-  infoIcon: '#0099FF',
-  infoText: '#FFFFFF',
-  warn: '#313132',
-  warnBorder: '#fcba19',
-  warnIcon: '#fcba19',
-  warnText: '#FFFFFF',
-  error: '#313132',
-  errorBorder: '#D8292F',
-  errorIcon: '#D8292F',
-  errorText: '#FFFFFF',
+  success: '#DFF0D8',
+  successBorder: '#D6E9C6',
+  successIcon: '#2d4821',
+  successText: '#2d4821',
+  info: '#D9EAF7',
+  infoBorder: '#B9CEDE',
+  infoIcon: '#313132',
+  infoText: '#313132',
+  warn: '#F9F1C6',
+  warnBorder: '#FAEBCC',
+  warnIcon: '#6C4A00',
+  warnText: '#6C4A00',
+  error: '#F2DEDE',
+  errorBorder: '#EBCCD1',
+  errorIcon: '#a12622',
+  errorText: '#a12622',
 }
 
 const GrayscaleColors: GrayscaleColors = {
@@ -192,40 +192,40 @@ interface TextBoxTheme {
 }
 
 export const Colors: ColorTheme = {
-  accent: BaseColors.yellow,
-  background: BaseColors.black,
-  backgroundLight: BaseColors.lightGreen,
-  borderLight: BaseColors.mediumGreen,
-  primary: BaseColors.darkGreen,
-  primaryActive: BaseColors.darkGreenHeavyTransparent,
+  accent: ColorPallet.brand.highlight,
+  background: ColorPallet.brand.primaryBackground,
+  backgroundLight: BaseColors.lightBlue,
+  borderLight: BaseColors.mediumBlue,
+  primary: ColorPallet.brand.primary,
+  primaryActive: BaseColors.darkBlueHeavyTransparent,
   shadow: BaseColors.darkGrey,
-  text: BaseColors.white,
+  text: BaseColors.darkGreyAlt,
   ...BaseColors,
   ...StatusColors,
 }
 
 export const CredentialOfferTheme: CredentialOfferTheme = {
-  background: Colors.black,
+  background: Colors.white,
 }
 
 export const ContactTheme: ContactTheme = {
-  background: Colors.shadow,
+  background: Colors.white,
 }
 
 export const ProofRequestTheme: ProofRequestTheme = {
-  background: Colors.black,
+  background: Colors.white,
 }
 
 export const SettingsTheme: SettingsTheme = {
-  background: Colors.shadow,
+  background: Colors.white,
 }
 
 export const SingleSelectBlockTheme: SingleSelectBlockTheme = {
-  background: Colors.shadow,
+  background: Colors.white,
 }
 
 export const TextBoxTheme: TextBoxTheme = {
-  background: Colors.darkGreenLightTransparent,
+  background: Colors.darkBlueLightTransparent,
   border: Colors.borderLight,
   text: Colors.text,
 }
@@ -277,7 +277,7 @@ export const Buttons = StyleSheet.create({
   primaryDisabled: {
     padding: 16,
     borderRadius: 4,
-    backgroundColor: Colors.darkGreenHeavyTransparent,
+    backgroundColor: Colors.darkBlueHeavyTransparent,
   },
   primaryText: {
     ...TextTheme.normal,
@@ -302,7 +302,7 @@ export const Buttons = StyleSheet.create({
     padding: 16,
     borderRadius: 4,
     borderWidth: 2,
-    borderColor: Colors.darkGreenLightTransparent,
+    borderColor: Colors.darkBlueLightTransparent,
     backgroundColor: Colors.white,
   },
   secondaryText: {
@@ -314,7 +314,7 @@ export const Buttons = StyleSheet.create({
   secondaryTextDisabled: {
     ...TextTheme.normal,
     fontWeight: 'bold',
-    color: Colors.darkGreenLightTransparent,
+    color: Colors.darkBlueLightTransparent,
     textAlign: 'center',
   },
 })
