diff --git a/App/localization/en/index.ts b/App/localization/en/index.ts
index db7a3f3..848088b 100644
--- a/App/localization/en/index.ts
+++ b/App/localization/en/index.ts
@@ -159,7 +159,7 @@ const translation = {
   },
   "Screens": {
     "Splash": 'Splash',
-    "Onboarding": "Onboarding",
+    "Onboarding": "BC Wallet",
     "Terms": "Terms & Conditions",
     "CreatePin": 'Create 6-Digit Pin',
     "EnterPin": 'Enter Pin',
