diff --git a/node_modules/@aries-framework/core/build/modules/indy/services/IndyRevocationService.js b/node_modules/@aries-framework/core/build/modules/indy/services/IndyRevocationService.js
index e46e098..8283f90 100644
--- a/node_modules/@aries-framework/core/build/modules/indy/services/IndyRevocationService.js
+++ b/node_modules/@aries-framework/core/build/modules/indy/services/IndyRevocationService.js
@@ -121,10 +121,10 @@ let IndyRevocationService = class IndyRevocationService {
         if (!requestRevocationInterval.to) {
             throw new AriesFrameworkError_1.AriesFrameworkError(`Presentation requests proof of non-revocation with no 'to' value specified`);
         }
-        if ((requestRevocationInterval.from || requestRevocationInterval.from === 0) &&
-            requestRevocationInterval.to !== requestRevocationInterval.from) {
-            throw new AriesFrameworkError_1.AriesFrameworkError(`Presentation requests proof of non-revocation with an interval from: '${requestRevocationInterval.from}' that does not match the interval to: '${requestRevocationInterval.to}', as specified in Aries RFC 0441`);
-        }
+        // if ((requestRevocationInterval.from || requestRevocationInterval.from === 0) &&
+        //     requestRevocationInterval.to !== requestRevocationInterval.from) {
+        //     throw new AriesFrameworkError_1.AriesFrameworkError(`Presentation requests proof of non-revocation with an interval from: '${requestRevocationInterval.from}' that does not match the interval to: '${requestRevocationInterval.to}', as specified in Aries RFC 0441`);
+        // }
     }
 };
 IndyRevocationService = __decorate([
